src/riscv/linux/cpuinfo.c

```diff
@@ -0,0 +1,315 @@
+#include <inttypes.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <cpuinfo/log.h>
+#include <linux/api.h>
+#include <riscv/linux/api.h>
+
+/*
+ * Size, in chars, of the on-stack buffer used for parsing lines of /proc/cpuinfo.
+ * This is also the limit on the length of a single line.
+ */
+#define BUFFER_SIZE 2048
+
+static uint32_t parse_processor_number(
+        const char* processor_start,
+        const char* processor_end)
+{
+    const size_t processor_length = (size_t) (processor_end - processor_start);
+
+    if (processor_length == 0) {
+        cpuinfo_log_warning("Processor number in /proc/cpuinfo is ignored: string is empty");
+        return 0;
+    }
+
+    uint32_t processor_number = 0;
+    for (const char* digit_ptr = processor_start; digit_ptr != processor_end; digit_ptr++) {
+        const uint32_t digit = (uint32_t) (*digit_ptr - '0');
+        if (digit > 10) {
+            cpuinfo_log_warning("non-decimal suffix %.*s in /proc/cpuinfo processor number is ignored",
+                                (int) (processor_end - digit_ptr), digit_ptr);
+            break;
+        }
+
+        processor_number = processor_number * 10 + digit;
+    }
+
+    return processor_number;
+}
+
+static void parse_isa(
+        const char* isa_start,
+        const char* isa_end,
+        struct cpuinfo_riscv_linux_processor processor[restrict static 1])
+{
+    const size_t isa_length = (size_t) (isa_end - isa_start);
+    if (!(memcmp(isa_start, "rv32", 4) == 0 || memcmp(isa_start, "rv64", 4) == 0)) {
+        cpuinfo_log_error("Invalid isa format in /proc/cpuinfo: %.*s. It should start with either `rv32` or `rv64`",
+                          (int) (isa_length), isa_start);
+        return;
+    }
+
+    /* Mark the features as valid */
+    processor->flags |= CPUINFO_RISCV_LINUX_VALID_FEATURES | CPUINFO_RISCV_LINUX_VALID_PROCESSOR;
+
+    const char* feature = isa_start + 4;
+    for (;feature != isa_end; feature++) {
+        if(*feature == 'a') {
+            #if CPUINFO_ARCH_RISCV32 || CPUINFO_ARCH_RISCV64
```

Comment:
Do we need this check? This function / this file requires us to be building for RISC-V.

Reply:
Removed checks, after verifying it still compiles on non-RISCV archs :)